### PR TITLE
Automatically open graphs if process.env.WEBPPL_VIZ_AUTOOPEN is set (resolves #86)

### LINKED ISSUE #86: Automatically open graphs
There should be an option to automatically open rendered graphs, when running webppl from command line.

The following patch to index.js does that by default:

```
@@ -3,6 +3,7 @@
 var $ = require('jquery');
 var dependencyAnalysis = require('./dependency-analysis');
 var reflection = require('./reflection');
+var open = require("open");
 
 global.d3 = d3;
 
@@ -1039,6 +1040,7 @@
 
                     require('fs').writeFileSync(fileName, svgText);
                     console.log("Rendered to " + fileName);
+                    open(fileName)
                     util.trampolineRunners.cli()( options.callback() )
                   });
 
```

### PR DESCRIPTION
**Pull request by smihael**
You have to run `export WEBPPL_VIZ_AUTOOPEN="true"` for this to work.
Resolves bug #86

### CODE REVIEW

**Comment by smihael** [comment]
I also changed naming of graphs in rev 95b13b6 from md5 hash into date-time format to avoid confusion, but this is not essential.

File: src/index.js [hunk 1]
```diff
@@ -1036,12 +1036,23 @@ function renderSpec(spec, _options) {
...
-                    var fileName = options.fileName || (md5(svgText).substring(0,7) + '.svg');
+                    
+                    var now = new Date();
+                    var str = now.getUTCFullYear().toString() + "_" +
+                            (now.getUTCMonth() + 1).toString() + "_" +
+                            now.getUTCDate() + "_" +
+                            now.getUTCHours() + "_" +
+                            now.getUTCMinutes() + "_" +
+                            now.getUTCSeconds() + "_" +
+                            now.getMiliseconds;
```
Comment (longouyang) [inline src/index.js]: Good idea but there is a typo here. Also, I imagine we probably don't need millisecond precision in the names anyway, some maybe just remove this.

**Review by longouyang — CHANGES_REQUESTED** [review]
Just a small fix required.

Also, do you know if this works on Windows? (not required, I'm just curious)

File: src/index.js (hunk 1 above)
Comment (smihael) [reply src/index.js]: I'd stick to milliseconds. Imagine you run viz(repeat(100,flip)) directly after viz(repeat(10,flip)). Both commands get executed within single second and the file would get overwritten. To cope with that mills are appended to the filename only if the file exists already.

**Comment by smihael** [comment]
I fixed the typo. As regards Windows support, I can't test on my own, but according to https://www.twilio.com/blog/2017/01/how-to-set-environment-variables.html working with environment variables in Node.js should be possible on Windows as well using the sysdm.cpl tool.

**Review by longouyang — APPROVED** [review]

**Comment by smihael** [comment]
Can you also add cfacec3ac8c8562778506720147d83d9ff22985e from my fork?

**Comment by longouyang** [comment]
I think I wrote it that way to avoid calling `require('fs')` inside the browser, because I wasn't sure that would work.

If you can do some testing and determine that declaring `var fs = require('fs')` doesn't break the browser build, then I can add it.

**Pull request merged**